arch/tricore/src/common/Toolchain.defs

```diff
@@ -0,0 +1,127 @@
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+#
+# Supported toolchains
+#
+# Each toolchain definition should set:
+#
+#  CROSSDEV         The GNU toolchain triple (command prefix)
+#  ARCHCPUFLAGS     CPU-specific flags selecting the instruction set
+#                   FPU options, etc.
+#  ARCHOPTIMIZATION The optimization level that results in
+#                   reliable code generation.
+#
+
+ifeq ($(CONFIG_DEBUG_CUSTOMOPT),y)
+  ARCHOPTIMIZATION += $(CONFIG_DEBUG_OPTLEVEL)
+else ifeq ($(CONFIG_DEBUG_FULLOPT),y)
+  ifeq ($(CONFIG_ARCH_TOOLCHAIN_CLANG),y)
+    ARCHOPTIMIZATION += -Oz
+  else
+    ARCHOPTIMIZATION += -Os
+  endif
+endif
+
+# Tasking toolchain
+
+CC                = cctc
+CXX               = cctc
+CPP               = cctc $(ARCHOPTIMIZATION)
+LD                = cctc
+STRIP             = strip --strip-unneeded
+AR                = artc -r
+NM                = nm
+OBJCOPY           = echo
+OBJDUMP           = elfdump
+
+ARCHOPTIMIZATION += --lsl-core=vtc
+LDFLAGS          += --lsl-core=vtc
+ARCHOPTIMIZATION += --iso=99
+ARCHOPTIMIZATION += --language=+gcc,+volatile,-strings,-kanji
+
+# pragma align <4> (default: 0)
+
+ARCHOPTIMIZATION += --align=4
+
+# Always use 32-bit integers for enumeration
+
+ARCHOPTIMIZATION += --integer-enumeration
+
+# tradeoff between speed (-t0) and size (-t4) (default: 4)
+
+ARCHOPTIMIZATION += --tradeoff=2
+
+# enable symbolic debug information
+
+ifeq ($(CONFIG_DEBUG_SYMBOLS),y)
+  ARCHOPTIMIZATION += --debug-info=default
+  ARCHOPTIMIZATION += --keep-temporary-files
+  LDFLAGS          += -g
+endif
+
+# merge source code with assembly output
+
+ARCHOPTIMIZATION += --source
+
+# generate alignment depending on assume_if hints
+
+ARCHOPTIMIZATION += --branch-target-align
+
+
+# Since nuttx uses too many of GNU extensions in the implementation of
+# FPU-related library functions, which is not supported in tasking,
+# so currently we cannot use FPU-related configurations to manage it.
+#
+# Just set fp-model to Double Precision:
+# --fp-model[=<flag>,...]         floating-point model (default: cFlnrSTz)
+#   0                               alias for --fp-model=CFLNRStZ (strict)
+#   1                               alias for --fp-model=cFLNRSTZ (precise)
+#   2                               alias for --fp-model=cFlnrSTz (fast-dp)
+#   3                               alias for --fp-model=cflnrSTz (fast-sp)
+
+ARCHOPTIMIZATION += --fp-model=2
+LDFLAGS          += --fp-model=2
+LDFLAGS          += -lfp_fpu
+
+LDFLAGS          += --hex-format=s -Wl-OtxYcL -Wl-mcrfiklsmnoduq
+LDFLAGS          += -lrt
+
+# ctc W500: ["stdio/lib_libvsprintf.c" 884/29] expression without effect
```

Comment:
should we fix these warning instead disabling them

Reply:
Most of the warnings are compiler defects, which have been fixed in GNU toolchain